opentelemetry-exporter-gcp-logging/src/opentelemetry/exporter/cloud_logging/__init__.py

```diff
@@ -0,0 +1,258 @@
+from google.cloud.logging_v2.services.logging_service_v2 import (
+    LoggingServiceV2Client,
+)
+from google.cloud.logging_v2.services.logging_service_v2.transports.grpc import (
+    LoggingServiceV2GrpcTransport,
+)
+from google.cloud.logging_v2.types.log_entry import LogEntry
+from google.cloud.logging_v2.types.logging import WriteLogEntriesRequest
+from google.logging.type.log_severity_pb2 import LogSeverity  # type: ignore
+from google.protobuf.struct_pb2 import Struct
+from google.protobuf.timestamp_pb2 import Timestamp
+from opentelemetry.exporter.cloud_logging.version import __version__
+from opentelemetry.resourcedetector.gcp_resource_detector._mapping import (
+    get_monitored_resource,
+)
+from opentelemetry.sdk import version as opentelemetry_sdk_version
+from opentelemetry.sdk._logs import LogData
+from opentelemetry.sdk._logs.export import LogExporter
+from opentelemetry.sdk.resources import Resource
+from opentelemetry.trace import format_span_id, format_trace_id
+
+DEFAULT_MAX_ENTRY_SIZE = 256000  # 256 KB
+DEFAULT_MAX_REQUEST_SIZE = 10000000  # 10 MB
+
+HTTP_REQUEST_ATTRIBUTE_KEY = "gcp.http_request"
+LOG_NAME_ATTRIBUTE_KEY = "gcp.log_name"
+SOURCE_LOCATION_ATTRIBUTE_KEY = "gcp.source_location"
+TRACE_SAMPLED_ATTRIBUTE_KEY = "gcp.trace_sampled"
+PROJECT_ID_ATTRIBUTE_KEY = "gcp.project_id"
+_OTEL_SDK_VERSION = opentelemetry_sdk_version.__version__
+_USER_AGENT = f"opentelemetry-python {_OTEL_SDK_VERSION}; google-cloud-logging-exporter {__version__}"
+
+# Set user-agent metadata, see https://github.com/grpc/grpc/issues/23644 and default options
+# from
+# https://github.com/googleapis/python-logging/blob/5309478c054d0f2b9301817fd835f2098f51dc3a/google/cloud/logging_v2/services/logging_service_v2/transports/grpc.py#L179-L182
+_OPTIONS = [
+    ("grpc.max_send_message_length", -1),
+    ("grpc.max_receive_message_length", -1),
+    ("grpc.primary_user_agent", _USER_AGENT),
+]
+
+# severityMapping maps the integer severity level values from OTel [0-24]
+# to matching Cloud Logging severity levels.
+SEVERITY_MAPPING: dict[int, int] = {
+    0: LogSeverity.DEFAULT,
+    1: LogSeverity.DEBUG,
+    2: LogSeverity.DEBUG,
+    3: LogSeverity.DEBUG,
+    4: LogSeverity.DEBUG,
+    5: LogSeverity.DEBUG,
+    6: LogSeverity.DEBUG,
+    7: LogSeverity.DEBUG,
+    8: LogSeverity.DEBUG,
+    9: LogSeverity.INFO,
+    10: LogSeverity.INFO,
+    11: LogSeverity.NOTICE,
+    12: LogSeverity.NOTICE,
+    13: LogSeverity.WARNING,
+    14: LogSeverity.WARNING,
+    15: LogSeverity.WARNING,
+    16: LogSeverity.WARNING,
+    17: LogSeverity.ERROR,
+    18: LogSeverity.ERROR,
+    19: LogSeverity.ERROR,
+    20: LogSeverity.ERROR,
+    21: LogSeverity.CRITICAL,
+    22: LogSeverity.CRITICAL,
+    23: LogSeverity.ALERT,
+    24: LogSeverity.EMERGENCY,
+}
+
+
+def convert_any_value_to_string(value: Any) -> str:
+    t = type(value)
+    if t is bool or t is int or t is float or t is str:
+        return str(value)
+    if t is list or t is tuple:
```

Comment:
better to use ABCs and `isinstance()` checks so it works with subclasses too. feel free to follow up